src/Generator.php

```diff
@@ -24,13 +25,13 @@ class Generator
     private $fileManager;
     private $twigHelper;
     private $pendingOperations = [];
-    private $namespacePrefix;
+    private $namespacesHelper;
 
-    public function __construct(FileManager $fileManager, string $namespacePrefix)
+    public function __construct(FileManager $fileManager, NamespacesHelper $namespacesHelper)
```

Comment:
For backwards compatibility, we need to temporarily allow for both. You would:

A) Remove the type-hint
B) If it's a string, use `trigger_error` to trigger a deprecation. The, manually create a NamespacesHelper with all the default values.

Reply:
@weaverryan I've implemented the solution you've suggested and I'm also checking the type of the provided `$namespacesHelper` to make sure either a string or a NamespacesHelper instance is given because without the type hint it is the only way we have to enforce it.

Also please have a look at the exception and `trigger_error` messages because I wasn't sure about what to write :)